app/views/menus/show.html.haml

```diff
@@ -12,5 +12,5 @@
           %ul
             %li
               = link_to dish.name, dish
-      - if can? :manage, Menu
+      - if (can? :manage, Menu) && (current_user = @menu.restaurant.user)
```

Comment:
Same here, if the ability is set up correctly, `can? :manage, Menu` should be enough to prevent unauthorized access.